pkg/action/reset.go

```diff
@@ -113,24 +113,35 @@ func (r ResetAction) Run() (err error) {
 	cleanup := utils.NewCleanStack()
 	defer func() { err = cleanup.Cleanup(err) }()
 
-	// Unmount partitions if any is already mounted before formatting
-	// TODO: Is this needed???
-	err = e.UnmountPartitions(r.spec.Partitions.PartitionsByMountPoint(true, r.spec.Partitions.Recovery))
-	if err != nil {
-		return err
-	}
-
 	// Reformat state partition
-	err = e.FormatPartition(r.spec.Partitions.State)
-	if err != nil {
-		return err
-	}
+	// We should expose this under a flag, to reformat state before starting
+	// In case state fs is broken somehow
+	/*
+		if r.spec.FormatState {
+			state := r.spec.Partitions.State
+			if state != nil {
+				err = e.UnmountPartition(state)
+				if err != nil {
+					return err
+				}
+				err = e.FormatPartition(state)
+				if err != nil {
+					return err
+				}
+				// Mount it back
+				err = e.MountPartition(state)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	*/
 
 	// Reformat persistent partition
 	if r.spec.FormatPersistent {
 		persistent := r.spec.Partitions.Persistent
 		if persistent != nil {
-			err = e.UnmountPartitions(r.spec.Partitions.PartitionsByMountPoint(true, persistent))
```

Comment:
didn't the `excludes` flag work? (In `PartitionsByMountPoint`). Maybe that method needs fixing then?

Reply:
wait, we are doing the opposite now? We use to unmount "all but persistent" and now we unmount "persistent" only?

Reply:
Yes, this was wrong, it worked and we were unmounting everything but persistent for some reason. We wanted the other way, before formatting persistent we want to unmount persistent.

Reply:
:upside_down_face: